### ISSUE TITLE
Deduplicating Bikeshed metadata

### ISSUE THREAD

**Issue by domenic** [whatwg/meta#23]
Here is my vision:

```
Group: WHATWG
Title: Streams Standard
H1: Streams
Shortname: streams
Text Macro: TWITTER streamsstandard
Abstract: ...
```

I would like this to generate the following additional metadata:

```
Repository: whatwg/streams
Status: LS
Boilerplate: omit conformance, omit feedback-header
No Editor: true
Logo: https://resources.whatwg.org/logo-streams.svg
!Participate: <a href="https://github.com/whatwg/streams">GitHub whatwg/streams</a> (<a href="https://github.com/whatwg/streams/issues/new">new issue</a>, <a href="https://github.com/whatwg/streams/issues">open issues</a>)
!Participate: <a href="https://wiki.whatwg.org/wiki/IRC">IRC: #whatwg on Freenode</a>
!Commits: <a href="https://github.com/whatwg/streams/commits">GitHub whatwg/streams/commits</a>
!Commits: [SNAPSHOT-LINK]
!Commits: <a href="https://twitter.com/streamsstandard">@streamsstandard</a>
!Tests: <a href="https://github.com/w3c/web-platform-tests/tree/master/streams">web-platform-tests streams/</a> (<a href="https://github.com/w3c/web-platform-tests/labels/streams">ongoing work</a>)
```

Almost all of this is inferred from the shortname.

Is that possible, @tabatkins? By editing defaults.include? The shortname substitution and the macro interaction I'm particularly unsure about.

**Comment by tabatkins** [comment]
I don't currently have a way to generate metadata from other metadata. (Metadata processing is the very first thing that happens, before anything else.)  But this sounds interesting!

Most likely it would have to be more complex than just dumb string substitution, which makes me lean towards adding a new Python hook for this.

**Comment by domenic** [comment]
OK, this seems to have worked pretty well; I finally got around to trying it. Here are the minor issues:

- https://github.com/tabatkins/bikeshed/issues/1057
- https://github.com/tabatkins/bikeshed/issues/1058
- DOM and MIME Sniffing still have legacy open bugs. DOM has so few we could migrate them, but MIME Sniffing has a lot. This ruins our predictable pattern for generating "Participate" links. Not sure what to do there; @annevk thoughts?
- Some specs don't have any tests (e.g. MIME Sniffing), so links to tests for them should not be output. Unsure what to do there; maybe we need more knobs in the Bikeshed feature?

Regarding https://github.com/tabatkins/bikeshed/issues/1058 and the above tests issue, here's one idea: You have to explicitly place your computed metadata. So e.g. my example from the OP would become

```
Group: WHATWG
Title: Streams Standard
H1: Streams
Shortname: streams
Text Macro: TWITTER streamsstandard
Abstract: ...
Computed Metadata Here: !Participate, !Commits, !Tests
!Demos: ...
!Translation (non-normative): ...
```

(Then, MIME Sniffing would do `Computed Metadata Here: !Participate, !Commits`)

I'm not sure that's the best solution, as it does add boilerplate to the spec (which the whole point of this issue is to avoid), but it definitely adds flexibility...

In general the conflation between metadata and headers is starting to feel a bit strange to me.

**Comment by annevk** [comment]
If we get a Python hook it should be easy to have some "has legacy issues" metadata in the input and generate the correct output.

**Comment by domenic** [comment]
This is in progress and seems to be working really nicely in https://github.com/whatwg/url/pull/360 + https://github.com/tabatkins/bikeshed/pull/1155.

My plan was just to update the existing WHATWG boilerplate in place then do a ton of pull requests all at once to all specs.

However we're somewhat blocked on https://github.com/whatwg/sg/issues/51 (which is a non-technical issue).

**Comment by domenic** [comment]
This has largely been done! https://github.com/whatwg/encoding/pull/132 is the lone holdout, but we can track that there. It's lovely! Thanks @tabatkins for being responsive to the small bugs I found as I finalized things.

**Comment by tabatkins** [comment]
I'm rather impressed at how minimal you were able to get the per-spec metadata!
